zenodo/modules/deposit/tasks.py

```diff
@@ -120,9 +120,14 @@ def cleanup_indexed_deposits():
          .query('term', **{'_deposit.status': 'draft'})
          .source(['_deposit.id']))
     res = q.scan()
-    es_depids_info = [(d.to_dict().get('_deposit.id', [None])[0], d.meta.id,
-                       d.meta.index, d.meta.doc_type) for d in res]
-    es_depids = {p[0] for p in es_depids_info}
+    erroneous_depids = es_depids_info = []
+    for d in res:
+        try:
+            es_depids_info.append((d.to_dict()['_deposit']['id'], d.meta.id,
+                                   d.meta.index, d.meta.doc_type))
+            es_depids = {p[0] for p in es_depids_info}
+        except KeyError:
```

Comment:
Maybe it's better to catch `Exception` here, to cover other unknown failures as well

Reply:
But do we really want to continue the task if we have a new Exception?

Reply:
Yup, it's about deposits that are deleted, but for some reason exist in the ES index (e.g. because they were postponed for bulk indexing) and are thus visible to users in their "Uploads" view. If they click on them they get a 404 page, and can't really do anything.